src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionFactory.cs

```diff
@@ -1105,6 +1106,34 @@ public override void Flush()
         {
         }
 
+        public override Task FlushAsync(CancellationToken cancellationToken) =>
+            Task.CompletedTask;
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) =>
+            WriteAsync(new ReadOnlyMemory<byte>(buffer, offset, count), cancellationToken).AsTask();
+
+#if NET
+        public override
+#else
+        private
+#endif
+        ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
+        {
+            EnsureNotDisposed();
+
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
Why not `ThrowIfCancellationRequested`? due to perf of exception?

Reply:
This method isn't `async`. The cancellation should be propagated as part of the task.